src/bokeh/models/renderers/glyph_renderer.py

```diff
@@ -148,6 +160,41 @@ def add_decoration(self, marking: Marking, node: Literal["start", "middle", "end
 
         return decoration
 
+    def construct_color_bar(self, visual: Literal["fill", "line"] = "fill", **kwargs) -> ColorBar:
+        ''' Construct and return a new ``ColorBar`` for this ``GlyphRenderer``.
+
+        Args:
+            visual ("fill" or "line", optional):
+                Which visual field to use for constructing a color bar (default: "fill")
+
+        Returns:
+            ColorBar
+
+        The constructed color bar will use the fill or line color mapper from
+        the  GlyphRenderer's main glyph, as specified by ``visual``. Extra
+        keyword arguments may be passed in to control ``ColorBar`` properties
+        such as `title`.
+
+        '''
+        from ...core.property.vectorization import Field
+        from ..annotations import ColorBar
+        from ..mappers import ColorMapper
+
+        if visual == "fill":
+            fill_color = self.glyph.fill_color
+            if not (isinstance(fill_color, Field) and isinstance(fill_color.transform, ColorMapper)):
+                raise ValueError("construct_color_bar expects fill_color to be a field with a colormapper transform")
+            return ColorBar(color_mapper=fill_color.transform, **kwargs)
+
+        elif visual == "line":
+            line_color = self.glyph.line_color
+            if not (isinstance(line_color, Field) and isinstance(line_color.transform, ColorMapper)):
+                raise ValueError("construct_color_bar expects line_color to be a field with a colormapper transform")
+            return ColorBar(color_mapper=line_color.transform, **kwargs)
```

Comment:
Technically these error checks do not cover every contingency. A user could replace a valid mapper with a plain column name or fixed value. We'd need a serialization validation check to catch that. But I've never seen anyone do that sort of thing, so I'm not too concerned.